From ef3507c482a342661ce8075f78359e653a4486cd Mon Sep 17 00:00:00 2001
From: Frederico Cadete <frederico.cadete@awtce.be>
Date: Thu, 17 Mar 2016 16:55:31 +0100
Subject: [PATCH] buildsystem: make example programs link to the generated
 protocols library

This avoids redefining the generation of protocol files
and fixes builds with parallel make (the header generation was missing).

Signed-off-by: Frederico Cadete <frederico.cadete@awtce.be>
---
 .../multi-touch-viewer/CMakeLists.txt                 | 10 +---------
 .../simple-ivi-share/CMakeLists.txt                   | 19 +------------------
 2 files changed, 2 insertions(+), 27 deletions(-)

diff --git a/ivi-layermanagement-examples/multi-touch-viewer/CMakeLists.txt b/ivi-layermanagement-examples/multi-touch-viewer/CMakeLists.txt
index 42f7d22..87623fe 100644
--- a/ivi-layermanagement-examples/multi-touch-viewer/CMakeLists.txt
+++ b/ivi-layermanagement-examples/multi-touch-viewer/CMakeLists.txt
@@ -25,14 +25,6 @@ project (generate-protocol-src)
 
 find_program(WAYLAND_SCANNER_EXECUTABLE NAMES wayland-scanner)
 
-add_custom_command(
-    OUTPUT  ivi-application-protocol.c
-    COMMAND ${WAYLAND_SCANNER_EXECUTABLE} code
-            < ${CMAKE_SOURCE_DIR}/protocol/ivi-application.xml
-            > ${CMAKE_CURRENT_BINARY_DIR}/ivi-application-protocol.c
-    DEPENDS ${CMAKE_SOURCE_DIR}/protocol/ivi-application.xml
-)
-
 ################################################################
 
 project (multi-touch-viewer)
@@ -61,6 +53,7 @@ link_directories(
 
 SET(LIBS
     ${LIBS}
+    ivi-extension-protocol
     ${GLESv2_LIBRARIES}
     ${WAYLAND_CLIENT_LIBRARIES}
     ${WAYLAND_EGL_LIBRARIES}
@@ -82,7 +75,6 @@ set(HEADER_FILES
 
 add_executable(${PROJECT_NAME}
                ${SRC_FILES}
-               ${CMAKE_CURRENT_BINARY_DIR}/ivi-application-protocol.c
 )
 
 add_dependencies(${PROJECT_NAME} ${LIBS})
diff --git a/ivi-layermanagement-examples/simple-ivi-share/CMakeLists.txt b/ivi-layermanagement-examples/simple-ivi-share/CMakeLists.txt
index 248eafd..7401d08 100644
--- a/ivi-layermanagement-examples/simple-ivi-share/CMakeLists.txt
+++ b/ivi-layermanagement-examples/simple-ivi-share/CMakeLists.txt
@@ -25,22 +25,6 @@ project (generate-protocol-src)
 
 find_program(WAYLAND_SCANNER_EXECUTABLE NAMES wayland-scanner)
 
-add_custom_command(
-    OUTPUT  ivi-application-protocol.c
-    COMMAND ${WAYLAND_SCANNER_EXECUTABLE} code
-            < ${CMAKE_SOURCE_DIR}/protocol/ivi-application.xml
-            > ${CMAKE_CURRENT_BINARY_DIR}/ivi-application-protocol.c
-    DEPENDS ${CMAKE_SOURCE_DIR}/protocol/ivi-application.xml
-)
-
-add_custom_command(
-    OUTPUT  ivi-share-protocol.c
-    COMMAND ${WAYLAND_SCANNER_EXECUTABLE} code
-            < ${CMAKE_SOURCE_DIR}/protocol/ivi-share.xml
-            > ${CMAKE_CURRENT_BINARY_DIR}/ivi-share-protocol.c
-    DEPENDS ${CMAKE_SOURCE_DIR}/protocol/ivi-share.xml
-)
-
 ################################################################
 
 project (simple-ivi-share)
@@ -69,6 +53,7 @@ link_directories(
 
 SET(LIBS
     ${LIBS}
+    ivi-extension-protocol
     ${GLESv2_LIBRARIES}
     ${WAYLAND_CLIENT_LIBRARIES}
     ${WAYLAND_EGL_LIBRARIES}
@@ -82,8 +67,6 @@ SET(SRC_FILES
 
 add_executable(${PROJECT_NAME}
                ${SRC_FILES}
-               ${CMAKE_CURRENT_BINARY_DIR}/ivi-application-protocol.c
-               ${CMAKE_CURRENT_BINARY_DIR}/ivi-share-protocol.c
 )
 
 add_dependencies(${PROJECT_NAME} ${LIBS})
-- 
1.9.1

